OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -2876,6 +2876,22 @@ internal static void UpgradeActorRules(int engineVersion, ref List<MiniYamlNode>
 					}
 				}
 
+				if (engineVersion < 20160109)
+				{
+					//Divide levels by 100
+					if (depth == 1 && (node.Key == "GainsExperience" || node.Key.StartsWith("GainsExperience@"))
+							&& node.Value.Nodes.FirstOrDefault(n => n.Key == "Upgrades") != null)
+					{
+						foreach (var kv in node.Value.Nodes.Where(n => n.Key == "Upgrades")) {
+							var key = float.Parse(kv.Key) / 100;							
+							// t is a MiniYamlNode that looks like ("200", "firepower, damage, speed, reload, inaccuracy, rank")
+							// The key of this needs to be divided by 100 and saved.
+							var newKv = new MiniYamlNode(key.ToString(), kv.Value);
```

Comment:
I also don't know how else to solve it. There is only one way to be sure. Run the upgrade code on the pull request.